doc/user-guide/push-notifications/Push-notifications.md

```diff
@@ -0,0 +1,178 @@
+## How to set up Push Notifications
+
+Push notifications are the bread and butter of the modern mobile experience, and MongooseIM has
+support for them. When used together with [MongoosePush][], you get out-of-the-box push
+notifications for FCM (**F**irebase **C**loud **M**essaging) and APNS (**A**pple **P**ush
+**N**otification **S**ervice) providers. And it's easy to extend it to any other protocols of your
+choice.
+
+Server side push notification support is fully compliant with [XEP-0357 Push Notifications][XEP-0357],
+which defines several components that need to work together in order to provide clients with working
+push notifications. However, there's just one non-optimal detail required by the aforementioned XEP:
+that push notifications being a PubSub service — [we can do better than that][pubsub-less].
+
+If you're already familiar with the workings of [XEP-0357][], make sure to have a look at
+[our pubsub-less enhancement][pubsub-less].
+
+You might also want to read about the push notification's [client side configuration][client-side].
+
+All push notification mechanisms use [mod_event_pusher_push][] as a backend implementation,
+read [this][mod_event_pusher] to know more about it.
+
+While the whole setup can be incredibly extensible, we see the following straightforward uses of it.
+
+### XEP-0357 compliant with local PubSub
+
+ ![](PushNotifications-MIM_with_local_pubsub.png)
+
+It allows your clients to enable push notifications via a local PubSub.
+```erlang
+{mod_pubsub, [{plugins, [<<"push">>]}]}, % mandatory minimal config
+{mod_event_pusher, [
+    {backends, [
+        {push, [
+            {backend, mnesia}, % optional
+            {wpool, [{workers, 200}]}, % optional
+            {plugin_module, mod_event_pusher_push_plugin_defaults} % optional
+        ]}
+    ]}
+]}
+```
+
+### XEP-0357 compliant with remote PubSub
+
+ ![](PushNotifications-MIM_with_remote_pubsub.png)
+
+In this case, clients must provide a PubSub service domain that exists on another XMPP cluster.
+Notifications will be delivered over XMPP Federation to the remote server(s), so it all comes down to a
+[federation configuration][federation].
+
+#### Advantages
+* The whole responsibility is outsourced, MongooseIM will solely route the stanza abroad.
+* No need to worry about pubsub-less/pubsub-full or [XEP-0357][] compliance, that's precisely
+  outsourced.
+
+#### Drawbacks
+* It still all depends on how the federated server is configured. It can be a transparent custom
+  service, or another MongooseIM node with any of the previous configurations.
+
+If MongooseIM is used as the remote PubSub node, the architecture then looks like this:
+ ![](PushNotifications-MIM_as_remote_pubsub.png)
+
+The components that make push notifications possible in MongooseIM in the PubSub-full setup
+add up to the following architecture:
+![](component-architecture-with-PubSub.png)
```

Comment:
There is a typo in the file the link refers to. It contains `achitecture` instead of `architecture`, so the link does not work.